wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/monitoring/MonitoringRdsHostListProvider.java

```diff
@@ -89,41 +77,46 @@ public static void clearCache() {
     clearAll();
   }
 
-  public static void closeAllMonitors() {
-    monitors.getEntries().values().forEach(monitor -> {
-      try {
-        monitor.close();
-      } catch (Exception ex) {
-        // ignore
-      }
-    });
-    monitors.clear();
-    clearCache();
-  }
-
   @Override
   protected void init() throws SQLException {
     super.init();
   }
 
-  protected ClusterTopologyMonitor initMonitor() {
-    return monitors.computeIfAbsent(this.clusterId,
-        (key) -> new ClusterTopologyMonitorImpl(
-            key, topologyCache, this.initialHostSpec, this.properties, this.pluginService,
-            this.hostListProviderService, this.clusterInstanceTemplate,
-            this.refreshRateNano, this.highRefreshRateNano, TOPOLOGY_CACHE_EXPIRATION_NANO,
+  protected ClusterTopologyMonitor initMonitor() throws SQLException {
+    return monitorService.runIfAbsent(
+        ClusterTopologyMonitorImpl.class,
+        this.clusterId,
+        this.storageService,
+        this.pluginService.getTelemetryFactory(),
+        this.originalUrl,
+        this.pluginService.getDriverProtocol(),
+        this.pluginService.getTargetDriverDialect(),
+        this.pluginService.getDialect(),
+        this.properties,
+        (ConnectionService connectionService, PluginService monitorPluginService) -> new ClusterTopologyMonitorImpl(
+            this.clusterId,
+            this.storageService,
+            this.monitorService,
+            connectionService,
+            this.initialHostSpec,
+            this.properties,
+            monitorPluginService,
+            this.serviceContainer.getHostListProviderService(),
+            this.clusterInstanceTemplate,
+            this.refreshRateNano,
+            this.highRefreshRateNano,
             this.topologyQuery,
             this.writerTopologyQuery,
-            this.nodeIdQuery),
-        MONITOR_EXPIRATION_NANO);
+            this.nodeIdQuery));
   }
 
   @Override
   protected List<HostSpec> queryForTopology(final Connection conn) throws SQLException {
-    ClusterTopologyMonitor monitor = monitors.get(this.clusterId, MONITOR_EXPIRATION_NANO);
+    ClusterTopologyMonitor monitor = monitorService.get(ClusterTopologyMonitorImpl.class, this.clusterId);
```

Comment:
Should we use ClusterTopologyMonitorImpl.class or ClusterTopologyMonitor.class?

Reply:
I think it makes sense to use the concrete class because we might have other concrete implementations of ClusterTopologyMonitor in the future